docs/development/designs/automationGo.md

```diff
@@ -0,0 +1,49 @@
+# Problem
+
+Maintaining unsupported versions of Kubernetes and by extention Golang/Go requires large amounts of monitoring, evaluation, occassional deep dives into the functionality and differences in versions, and large amounts of repetitive searchable documentation for the Open-Source Community and transparency.
+
+Specifically for patching golang, we are required to monitor their announcements for vulnerabilities, open our own CVE issues for backports that are consistent to allow for easily searchable auditing and transparency for the open-source community, attempt backporting the fixes, and when the backport doesn't apply cleanly, discover the differences in versions to make the backport apply, if applicable.
+
+While the last task will require effort by the team, the rest of the task can be automated to provide clear and consistent [unknown phrasing/word here]. An automated solution would build a consistant audit trail for which CVEs have been patched or why they weren't, any public discussion on the decision process, and many other data points that could be of interest.
+
+# Solution Overview
+
+I prepose creating a go application for patching. Starting with Golang and potentially extending the functionality to Kubernetes as the workflow should be similar.
+
+## Golang Patching Runbook
+
+1. Golang release CVE announcement
+2. Open cve backport issues for EKS supported versions
+3. Attempt cherry-pick
+   3a. Find upstream commit fix
+   3b. git cherry-pick {commit hash}
+   3c. fix any merge issues
+4. Create .patch file with EKS patch header
+5. Create pr with .patch file fix and updated .spec file
+   (optional) upload additional source files required for patch
+6. Presubmit: pass/fail
+   6a (pass). Merge PR.
+   6b (fail). Repeat steps 3c - 6 with attempted fixes for build failures
+
+## Expected Kubernetes Patching Runbook
+
+Similar to Golang
+
+## Deep Dive
+
+Use a prow job to a call the cve search Golang repo for CVE [backport] issue. Once cve backport issue is created, trigger a job to open AWS/EKS-Distro-Build-Tooling issues for eks supported versions of go. Once these issues are created, watch the upstream issue for closed by [commit] comment. Use this commit to cherry-pick, format-patch, and submit initial PR is no merge conflicts. If merge conflict, create comment in the issue that a pr couldn't be autogenerated.
```

Comment:
My only hesistation here is this is brittle if upstream doesn't follow the exact same process each time for every CVE (I've already seen one issue that was added after the fact and didn't follow the expected format), and if the process changed altogether we'd have to see that and update in time. 

We can discuss how to handle. I can see it still trying to open PRs and issues automaticaly, but if we also ensure each time a CVE is released we have a human looking to see that the issues were properly created that should do it. Or just skip the looking bit and pass it an upstream issue and a commit hash and let the tool take it from there.

Reply:
I think it'd be good to automatically monitor the upstream releases and open new issues when a new release occurs with a link to the release announcement, even if it's not directly tied to the rest of the automation. Would be a good way to stay on top of new Go releases.

We could then use that issue as the base-issue for all the CVE backports and proceed with chatops commands like '/backport $COMMIT_ID $CVE_ID $RELEASE_BRANCH' or w/e we want?

Reply:
I agree some monitoring would be beneficial even if it wasn't directly tied to kicking off the automation. 

Also, I think we could mitigate some of the issues with searching for some of their automated messages that are created and documented in their gopherbot. There cherrypick and backport comments come from their custom bot which should make searching a bit more consistent.